system_setup/server/controllers/configure.py

```diff
@@ -37,6 +40,90 @@ def __init__(self, app):
     def start(self):
         self.install_task = self.app.aio_loop.create_task(self.configure())
 
+    def _update_locale_gen(self, localeGenPath, lang):
+        """ Uncomments the line in locale.gen file where lang is found,
+            if found commented. A fully qualified language is expected,
+            since that would have passed thru the Locale
+            controller validation. e.g. en_UK.UTF-8. """
+
+        fileContents: str
+        locGenNeedsWrite = False
+        with open(localeGenPath, "r") as f:
+            fileContents = f.read()
+            lineFound = fileContents.find(lang)
+            if lineFound == -1:
+                # An unsupported locale coming from our UI is a bug
+                log.error("Selected language %s not supported.", lang)
+                return
```

Comment:
Maybe we should return false to stop parent on error cases? (same for all error cases below)

Reply:
Nice tip. Applied across the board.

> And here we go, thanks for pass 2! I think there are still opportunities to get part of the code (outside of the mock question) better, but this is going in the right direction IMHO.
> 
> Feel free to tell me if you disagree with anything written here or if things aren’t clear!

You may not believe, but I really appreciate your critical thinking. It makes me think hard on ways to improve my code. Sometimes I get excited by the result and ended up not looking thoroughly enough to the way the result is achieved.

Thank you for that.